ONSdigital/census-rh-service
# Cr 1435 synchronous webform handling

# Motivation and Context
RM have refused to support our request for an additional queue in RabbitMQ, thus invalidating our current solution. It has been decided to do a simple synchronous call, but protect it with a circuit breaker.  

# What has changed
- remove webform event publishing
- add circuit breaker around call to GOV.UK notify
- the webform send to GOV.UK notify code is synchronous now from the endpoint call onwards
- if the GOV.UK notify site is down or slow then the RHSvc endpoint will return a failure response to the RHUI.
- additional Spring based circuit breaker tests. Focus on timeout handling.

# How to test?
- unit tests
- RH cucumber 
- call RHSvc endpoint using postman

To test the circuit-breaker handling manually RHSvc can be started with customised environment variables for the notify and webform circuit breaker parameters to see circuit breaker in action.  If a "mock notify" is constructed with a specific delay then the timeout can be tested, but this requires additional coding for a mock endpoint.

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1435
